data_prep/project_context/context_retriever.py

```diff
@@ -24,7 +26,8 @@ def __init__(self, project_name: str, function_signature: str):
     self._project_name = project_name
     self._function_signature = function_signature
     self._download_from_path = f'{self.AST_BASE_PATH}/{self._project_name}/*'
-    self._ast_path = f'{self.DOWNLOAD_TO_PATH}/{self._project_name}'
+    self._uuid = uuid.uuid4()
+    self._ast_path = f'{self.DOWNLOAD_TO_PATH}/{self._project_name}{self._uuid}'
```

Comment:
nit: add a `-` between `{self._project_name}` and `{self._uuid}` for visual simplicity.

Reply:
BTW, ideally we'd like to reuse the same data instead of re-downloading them.
But we can fix that when integrating FI later.

Reply:
added '-'. ack on reusing data